packages/studio/src/components/AddPageButton/EntityIdField.tsx

```diff
@@ -0,0 +1,125 @@
+  selectedIds?: string[];
+}
+
+export default function EntityIdField({
+  disabled,
+  updateSelection,
+  selectedIds,
+}: Props) {
+  const [entitiesRecord, fetchEntities] = useStudioStore((store) => [
+    store.accountContent.entitiesRecord,
+    store.accountContent.fetchEntities,
+  ]);
+  const availableEntityTypes = Object.entries(entitiesRecord)
+    .filter(([_entityType, entityData]) => {
+      return Object.values(entityData).length > 0;
+    })
+    .map((entry) => entry[0]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [entityType, setEntityType] = useState<string | undefined>(
+    availableEntityTypes[0]
+  );
+  const entityOptions = useEntityOptions(entityType);
+  const selectedOptions = useMemo(
+    () => selectedIds?.map((id) => ({ value: id, label: id })),
+    [selectedIds]
+  );
+
+  const onEntityTypeChange = useCallback(
+    (e: ChangeEvent<HTMLSelectElement>) => {
+      setEntityType(e.target.value);
+    },
+    []
+  );
+
+  const onEntityIdChange = useCallback(
+    (selectedItems: MultiValue<{ value: string }>) => {
+      updateSelection(selectedItems.map((item) => item.value));
+    },
+    [updateSelection]
+  );
+
+  const onMenuScrollToBottom = useCallback(async () => {
+    if (!entityType) {
+      return;
+    }
+    const currentNumEntities = Object.keys(entitiesRecord[entityType]).length;
+    if (currentNumEntities >= entitiesRecord[entityType].totalCount) {
+      return;
+    }
+    setIsLoading(true);
+    await fetchEntities(entityType, Math.floor(currentNumEntities / 50));
+    setIsLoading(false);
+  }, [entitiesRecord, entityType, fetchEntities]);
+
+  const className = classNames("flex flex-col mb-1");
+
+  if (availableEntityTypes.length === 0) {
+    return (
+      <div className={className}>
+        <StreamScopeFieldLabel streamScopeField="entityIds" />
+        <div className="border border-gray-400 min-h-[38px] rounded-lg pt-2 pb-1 px-2 w-full text-sm bg-gray-50 text-gray-500">
+          {!disabled && "No entity types found in the account."}
+        </div>
+      </div>
+    );
+  }
+
+  return (
+    <div className={className}>
+      <StreamScopeFieldLabel streamScopeField="entityIds" />
+      <label className="mt-2 text-sm flex flex-col">
+        {!disabled && "Choosing from entities of type:"}
+        <select
+          className={getSelectCssClasses()}
+          onChange={onEntityTypeChange}
+          disabled={disabled}
+        >
+          {!disabled &&
+            availableEntityTypes.map((entityType) => (
+              <option key={entityType}>{entityType}</option>
+            ))}
+        </select>
+      </label>
+      {!disabled && (
+        <Select
+          className="my-2"
+          isMulti={true}
+          onChange={onEntityIdChange}
+          value={selectedOptions}
+          options={entityOptions}
+          onMenuScrollToBottom={onMenuScrollToBottom}
+          isLoading={isLoading}
```

Comment:
Is it possible for the loading indicator to be at the bottom of the dropdown instead of in the search input? I think that would make it more clear that the user hasn't reached the end of the dropdown and more items are being loaded in

Reply:
I think we'd have to override things in react-select which I'd rather make another item. But I'd also like to be more incremental and wait for what product says here